app/youtube/model.go

```diff
@@ -0,0 +1,137 @@
+package youtube
+
+import (
+	"github.com/akali/steplems-bot/app/logger"
+	"github.com/kkdai/youtube/v2"
+	"io"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type (
+	Youtube struct {
+		pattern *regexp.Regexp
+		client  youtube.Client
+	}
+
+	VideoType   string
+	QualityType string
+)
+
+const (
+	HD     QualityType = "hd"
+	HD720  QualityType = "hd720"
+	HD1080 QualityType = "hd1080"
+	MEDIUM QualityType = "medium"
+)
+
+const (
+	MP4  VideoType = "video/mp4"
+	MKV  VideoType = "video/mkv"
+	WEBM VideoType = "video/webm"
+)
+
+var (
+	ytLinkRegex    = "(((?:https?:)?\\/\\/)?((?:www|m)\\.)?((?:youtube\\.com))(\\/(shorts\\/))([\\w\\-]+)(\\S+)?)"
+	log            = logger.Factory.NewLogger("youtube")
+	allowedQuality = []QualityType{HD, HD720, HD1080, MEDIUM}
+	allowedType    = []VideoType{MP4, MKV, WEBM}
+)
+
+func NewYoutube() *Youtube {
+	r := regexp.MustCompile(ytLinkRegex)
+
+	return &Youtube{
+		pattern: r,
+		client:  youtube.Client{},
+	}
+}
+
+func (yt *Youtube) Match(text string) []string {
+	return yt.pattern.FindAllString(text, -1)
+}
+
+func (yt *Youtube) chooseFormat(formats youtube.FormatList) *youtube.Format {
+	formats = formats.WithAudioChannels()
+
+	isFailedToFind := true
```

Comment:
No need for a separate flag, just set chosenFormat as formats[0] in the beginning

Reply:
Or even directly return format if you able to find, otherwise return &formats[0]